shard/src/test/java/com/redhat/developer/shard/ManagerSyncServiceTest.java

```diff
@@ -0,0 +1,114 @@
+package com.redhat.developer.shard;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.RequestListener;
+import com.github.tomakehurst.wiremock.http.RequestMethod;
+import com.github.tomakehurst.wiremock.http.Response;
+import com.redhat.developer.infra.dto.ConnectorDTO;
+import com.redhat.developer.infra.dto.ConnectorStatus;
+
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+
+@QuarkusTest
+@QuarkusTestResource(restrictToAnnotatedClass = true, value = ManagerMockResource.class)
+public class ManagerSyncServiceTest {
+
+    @Inject
+    ManagerSyncService managerSyncService;
+
+    @InjectWireMock
+    WireMockServer wireMockServer;
+
+    @BeforeEach
+    void cleanup() {
+        wireMockServer.resetAll();
+    }
+
+    @Test
+    public void testConnectorsAreDeployed() throws JsonProcessingException, InterruptedException {
+        List<ConnectorDTO> connectorDTOs = new ArrayList<>();
+        connectorDTOs.add(new ConnectorDTO("myId-1", "myName-1", "myEndpoint", "myCustomerId", ConnectorStatus.REQUESTED));
+        connectorDTOs.add(new ConnectorDTO("myId-2", "myName-2", "myEndpoint", "myCustomerId", ConnectorStatus.REQUESTED));
+        stubConnectorsToDeploy(connectorDTOs);
+        stubConnectorUpdate();
+        String expectedJsonUpdateRequest = "{\"id\": \"myId-1\", \"name\": \"myName-1\", \"endpoint\": \"myEndpoint\", \"customerId\": \"myCustomerId\", \"status\": \"PROVISIONING\"}";
+
+        CountDownLatch latch = new CountDownLatch(2); // Two updates to the manager are expected
+        addConnectorUpdateRequestListener(latch);
+
+        managerSyncService.fetchAndProcessConnectorsFromManager().await().atMost(Duration.ofSeconds(5));
+
+        Assertions.assertTrue(latch.await(30, TimeUnit.SECONDS));
```

Comment:
Anyway we can reduce this time for the tests? We're going to be running these a lot, so having to wait 30 seconds each time is going to add up over time.

Reply:
The time you see in this test is the maximum time we wait for the latch to go to `0`. So actually the test runs very fast, we never wait 30 seconds!

Reply:
OK thanks 👍